_posts/2020-06-12-solidity-0.6-inheritance.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+title:  "Solidity 0.6.x features: inheritance"
+date:   2020-06-12
+author: Elena Gesheva
+---
+
+Similar to object-oriented programming in Solidity - a contract-oriented language - the inheritance and polymorphism features are as widely adopted and critical for the language evolution. There is hardly any Solidity developer who hasn't used these language features in their contracts to decouple logic and increase code reuse. With version 0.6 of the language the main improvements introduced are to make already existing rules more explicit. The compiler continues to use C3 Linearization. See [Solidity documentation](https://solidity.readthedocs.io/en/latest/contracts.html?#inheritance) for inheritance and polymorphism definition and rules. From the new features introduced the more prominent are interface inheritance and multiple inheritance. We've examined all changes to existing and addition of these new functionalities in distinct groups below.
+
+## Explicitely `virtual` and `override`
+Functions are no longer virtual by default. This means a call to a non-virtual function will always execute that function regardless of any other contract in the inheritance hierarchy. This reduces abiguity that exists in version 0.5 where all functions are implicity virtual, allowing to be overriden further down the inheritance structure. This is especially dangerous in big inheritance graphs where this abiguity can lead to unintended behaviour and bugs.
+
+    pragma solidity ^0.5.17;
+    contract A {
+        uint public x;
+        function setValue(uint _x) public {
+            x = _x;
+        }
+    }
+
+    contract B {
+        uint public y;
+        function setValue(uint _y) public {
+            y = _y;
+        }
+    }
+
+    contract C is A, B {
+    }
```

Comment:
```suggestion
```solidity
    pragma solidity ^0.5.17;
    contract A {
        uint public x;
        function setValue(uint _x) public {
            x = _x;
        }
    }

    contract B {
        uint public y;
        function setValue(uint _y) public {
            y = _y;
        }
    }

    contract C is A, B {
    }
```
```